app/app.js

```diff
@@ -110,34 +127,31 @@ window.run = function() {
                     });
                 });
 
-                drawerController.selectItem('1');
+                Astro.registerRpcMethod(AppRpc.names.navigateToNewRootView, ['url', 'title'], function(res, url, title) {
+                    drawerController.navigateToNewRootView(url, title);
+                    res.send(null, 'success');
+                });
                 registerCanGoBackRpc(drawerController);
 
                 return drawerController;
             });
         };
 
-        var counterBadgeControllerPromise = CounterBadgeController.init(
-            HeaderConfig.cartHeaderContent.imageUrl,
-            HeaderConfig.cartHeaderContent.id
-        ).then(function(counterBadgeController) {
-            counterBadgeController.updateCounterValue(3);
-            return counterBadgeController;
-        });
-
-        welcomeModalControllerPromise.then(function(welcomeModalController) {
-            // The welcome modal can be configured to show only
-            // once -- on initial startup, by passing in the
-            // parameter `{forced: false}` below
-            welcomeModalController.show({forced: true});
-        });
-
         Application.getOSInformation().then(function(osInfo) {
             if (osInfo.os === Astro.platforms.ios && BaseConfig.iosUsingTabLayout) {
                 return createTabBarLayout(counterBadgeControllerPromise);
             }
             return createDrawerLayout(counterBadgeControllerPromise);
         }).then(function(menuController) {
+            // Show welcome modal only after layout is created for proper
+            // bookkeeping of the active view.
+            welcomeModalControllerPromise.then(function(welcomeModalController) {
```

Comment:
I would prefer to see a more flattened promise structure (not have a `.then` nested inside another `.then`). This can be achieved  by persisting `Application.getOSInformation().then(...)` into it's own promise and then `Promise.join`ing it to the `welcomeModalControllerPromise`. Also you will have to have a separate statement to handle the deep linking registration.